src/test/java/com/google/api/generator/engine/JavaCodeGeneratorTest.java

```diff
@@ -0,0 +1,874 @@
+        JavaDocComment.builder()
+            .addComment("Service Description: This is a test comment.")
+            .addComment("")
+            .addSampleCode("LibraryServiceStub libServiceStub = new LibraryServiceStub()")
+            .addOrderedList(
+                Arrays.asList(
+                    "A \"flattened\" method.",
+                    "A \"request object\" method.",
+                    "A \"callable\" method."))
+            .setDeprecated("This is a deprecated message.")
+            .build());
+  }
+
+  private CommentStatement createPreMethodLineComment(String commentString) {
+    return CommentStatement.withComment(LineComment.withComment(commentString));
+  }
+
+  private List<CommentStatement> createPreMethodJavaDocComment() {
+    return Arrays.asList(
+        CommentStatement.withComment(
+            JavaDocComment.builder()
+                .addComment("Add books to Shelf and check if there is a novel,")
+                .addComment("return string message as whether novel books are added to the shelf.")
+                .addComment("")
+                .addParam("shelf", "The Shelf object to which books will put.")
+                .addParam("stack", "The Stack of the BookKinds.")
+                .build()));
+  }
+
+  private MethodInvocationExpr shelfMapContainsKey(Variable map, List<Expr> arg) {
+    return MethodInvocationExpr.builder()
+        .setMethodName("containsKey")
+        .setExprReferenceExpr(VariableExpr.withVariable(map))
+        .setArguments(arg)
+        .setReturnType(TypeNode.BOOLEAN)
+        .build();
+  }
+
+  private MethodInvocationExpr putShelfToObj(Variable obj, String methodName, List<Expr> arg) {
+    return MethodInvocationExpr.builder()
+        .setMethodName(methodName)
+        .setExprReferenceExpr(VariableExpr.withVariable(obj))
+        .setArguments(arg)
+        .build();
+  }
+
+  private VariableExpr fieldFromShelfObject(Variable shelfVar, Variable field) {
+    return VariableExpr.builder()
+        .setVariable(field)
+        .setExprReferenceExpr(VariableExpr.withVariable(shelfVar))
+        .build();
+  }
+
+  private Variable createVarFromConcreteRef(ConcreteReference ref, String name) {
+    return Variable.builder().setName(name).setType(TypeNode.withReference(ref)).build();
+  }
+
+  private Variable createVarFromVaporRef(VaporReference ref, String name) {
+    return Variable.builder().setName(name).setType(TypeNode.withReference(ref)).build();
+  }
+
+  private Variable createVarFromType(TypeNode ref, String name) {
+    return Variable.builder().setName(name).setType(ref).build();
+  }
+
+  private PrimitiveValue createBooleanValue(String booleanValue) {
+    return PrimitiveValue.builder().setValue(booleanValue).setType(TypeNode.BOOLEAN).build();
+  }
+
+  private VariableExpr createVarDeclExpr(Variable v) {
+    return VariableExpr.builder().setVariable(v).setIsDecl(true).build();
+  }
+
+  private MethodDefinition overrideCreateBookMethod(
```

Comment:
createBookMethod for consistency please.

Reply:
createBookMethod is in use for abstract `createBook` method, and `overrideCreateBookMethod ` is for the overwritten `createBook` method in Novel class and anonymous class `new Book = {}`. Let me know if it makes sense. Thanks!